src/report/pullRequests.ts

```diff
@@ -148,4 +159,113 @@ export class PullRequestAnalytics {
 
     return averages;
   }
+
+  /** Returns the reviewer who gave the biggest amount of reviews per month */
+  getMonthlyReviewers(
+    reviews: PullRequestNode["reviews"]["nodes"],
+  ): PullRequestMetrics["reviewers"] {
+    if (reviews.length === 0) {
+      return [];
+    }
+    reviews.sort((a, b) =>
+      moment(a.submittedAt as string).diff(moment(b.submittedAt as string)),
+    );
+
+    // We get the month of the first date
+    let currentMonth = moment(reviews[0].submittedAt).startOf("month");
+
+    const monthsWithMatches: PullRequestMetrics["reviewers"] = [];
+    // let reviewsPerUser: Map<string, number> = new Map<string, number>();
+    let reviewsPerUser: { user: string; reviews: number; avatar: string }[] =
+      [];
+
+    for (const review of reviews) {
+      if (!review.submittedAt) {
+        this.logger.debug(
+          `Skipping review from ${review.author.login} as it is has a null date`,
+        );
+        continue;
+      }
+      // If it happened in the same month
+      if (currentMonth.diff(moment(review.submittedAt), "month") == 0) {
+        const reviewerIndex = reviewsPerUser
+          .map((u) => u.user)
+          .indexOf(review.author.login);
+        if (reviewerIndex > -1) {
+          // If the user exists, we increment his reviews
+          reviewsPerUser[reviewerIndex].reviews += 1;
+        } else {
+          // Else we push a new user
+          reviewsPerUser.push({
+            user: review.author.login,
+            reviews: 1,
+            avatar: review.author.avatarUrl,
+          });
+        }
+      } else {
+        // If the month is over, we check who reviewed the most that month
+        let topReviewer: { user: string; reviews: number; avatar: string } = {
+          user: "",
+          reviews: -1,
+          avatar: "",
+        };
+
+        for (const monthlyReviewer of reviewsPerUser) {
+          if (monthlyReviewer.reviews > topReviewer.reviews) {
+            topReviewer = monthlyReviewer;
+          }
+        }
+        // If there was at least one review, we add it to that month's top reviewer
+        if (topReviewer.reviews > 0) {
+          monthsWithMatches.push({
+            month: currentMonth.format("MMM YYYY"),
+            ...topReviewer,
+          });
+        }
+
+        // We move the month to the next one
+        currentMonth = moment(review.submittedAt).startOf("month");
+        // We reset the monthly review object
+        reviewsPerUser = [];
+        // We add a review to the current user
+        reviewsPerUser.push({
+          user: review.author.login,
+          reviews: 1,
+          avatar: review.author.avatarUrl,
+        });
+      }
+    }
+
+    return monthsWithMatches;
+  }
+
+  /** Returns the reviewer who gave the biggest amount of reviews */
+  getTopReviewer(
```

Comment:
topReviewer within which timeframe?

Reply:
In total.

We have two reviewers, the top of the month and the top in the repository's history.

Here is an example, where I'm the month's top reviewer, but you @mordamax are the top reviewer in the repo's history:
<details>
<summary>Opstooling metrics</summary>

<h3>Top reviewers</h3>

```mermaid
  gantt
    title Top reviewer per month
    dateFormat X
    axisFormat %s
    section Mar 2022
    mordamax : 0, 6
    section Apr 2022
    mordamax : 0, 1
    section May 2022
    mordamax : 0, 2
    section Jul 2022
    cmichi : 0, 2
    section Aug 2022
    cmichi : 0, 3
    section Sep 2022
    mordamax : 0, 1
    section Oct 2022
    BulatSaif : 0, 1
    section Jan 2023
    rzadp : 0, 17
    section Feb 2023
    mordamax : 0, 8
    section Mar 2023
    mordamax : 0, 2
    section Apr 2023
    mordamax : 0, 21
    section May 2023
    Bullrich : 0, 8
    section Jun 2023
    Bullrich : 0, 9
    section Jul 2023
    mutantcornholio : 0, 15
    section Aug 2023
    mordamax : 0, 8
    section Sep 2023
    mordamax : 0, 2
    section Oct 2023
    Bullrich : 0, 2
    section Nov 2023
    rzadp : 0, 1
    section Dec 2023
    Bullrich : 0, 2
    section Jan 2024
    Bullrich : 0, 2
  ```
<h4>Reviewer of the month</h4>
<img src="https://avatars.githubusercontent.com/u/8524599?u=b165ea45e9be2ad77c5f66ea46d9bb4427a754a5&v=4" alt="Bullrich's avatar">
@Bullrich with 2 reviews!
<h3>Top reviewer</h3>

<img src="https://avatars.githubusercontent.com/u/1177472?u=4a7990d66368c00a6a4b98f0de97462118211763&v=4" alt="mordamax's avatar">

@mordamax with a **total of 84 reviews**!<h1>Issues</h1>
</details>

Reply:
@Bullrich does it count the approvals or comments as well?
 I guess if it includes comments - then im testing a lot the command bot, which might have impacted that, but I'm not reviewing that much i think...

Reply:
It counts the review action, but not the comments in the review:
<img width="671" alt="image" src="https://github.com/paritytech/metrics/assets/8524599/2cb3d8a3-4cdb-4986-afb5-5c1734ae3a57">